report.md

```diff
@@ -0,0 +1,248 @@
+
+2. Are the functions just complex, or also long?
+
+Tianxing Wu: It is very long and complex
+
+John Söderholm: It is very long and quite complex.
+
+Angelica Engström: It is very long and complex
+
+Celina Tjärnström: It is complex and long.
+
+Tobias Hansson: Both
+ 
+3. What is the purpose of the functions?
+
+Tianxing Wu: the purpose is to try to infer types for a given source and target.
+
+Angelica Engström: The purpose is to check if the given method declaration is applicable with a value that has been used as a search/query term
+
+Celina Tjärnström: The purpose is to look at and compare method declarations and evaluate which is more specific, where variadic functions are less specific than others.
+
+John Söderholm: The purpose of the function is to find the most applicable in a list of method declarations given a set of additional parameters. 
+
+Tobias Hansson: The purpose of the function is the same as for Angelica, we picked functions that take input as different types, so they do things differently but we did not realise this until later.
+
+4. Are exceptions taken into account in the given measurements?
+
+Tianxing Wu: There is no exception in this function.
+
+Angelica Engström: There is no exception in this function.
+
+Celina Tjärnström: There is no exception in this function.
+
+John Söderholm: There is one direct exception in the function. It has been taken into account. 
+
+Tobias Hansson: There is one exception that was taken into account.
+
+5. Is the documentation clear w.r.t. all the possible outcomes?
+
+Tianxing Wu: There is no document for this function. It does not return anything, so it must have unknown side effects.
+
+John Söderholm: The documentation for this function primarily consists of in-line comments that are written before some of the conditional statements. However, the function does not have complete documentation in the form of JavaDoc. 
+
+Angelica Engström: The documentation for this function primarily consists of in-line comments that are written before some of the conditional statements. It also utilizes JavaDoc but it’s not 100% clear what is actually happening.
+
+Celina Tjärnström: The documentation for this function primarily consists of in-line comments that are written before some of the conditional statements. It does not use JavaDoc comments.
+
+Tobias Hansson: The java docs are unclear and confusing and do not tell what the function does. There are a lot of line comments that clarify what individual lines do, but it is still not explained what the whole function does.
+
+## Refactoring
+
+Plan for refactoring complex code:
+
+Tianxing Wu: Use Maybe functor (In java it is called Optional) to eliminate if statements.
+
+John Söderholm: Extracting complex if-statements into distinct functions.
```

Comment:
```suggestion
John Söderholm: Extracting complex if-statements into distinct functions. This would reduce the cyclomatic complexity significantly. One of these functions would have the responsibility to investigate and handle possible null parameters in the list of input methods.
```